From ffd8369b909c86653ef30593fddf67c9f5308583 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Mon, 5 Feb 2018 17:17:17 +0100
Subject: [PATCH] rhbz#1541486: Fix localized'ness of ooSetupFactoryUIName
 props

8f44fb5ee0bd21d192820ffd359504f6d1c68a52 "Mark Base window titles for
localisation" localized some uses of ooSetupFactoryUIName in
officecfg/registry/data/org/openoffice/Setup.xcu, but failed to mark the prop as
oor:localized="true" in officecfg/registry/schema/org/openoffice/Setup.xcs.  (So
some arbitrary localized version of the prop would win at runtime, potentially
displaying certain window titles in wrong languages.)

Other uses of ooSetupFactoryUIName in Setup.xcu (for "Calc", "Impress",
"Writer", etc.) have never been localized, and are probably not meant to be
localized.  AFAICS, irrespective of oor:localized="true" in Setup.xcs, their
values not being marked with xml:lang="en-US" in Setup.xcu will mean that they
will not appear in
translations/source/*/officecfg/registry/data/org/openoffice.po and will not
accidentally start to be translated after this commit.

(Theoretically, inconsistencies between oor:localized in xcs and uses of
xml:lang in xcu could statically be caught at build time, e.g., with
officecfg/util/sanity.xsl.  However, those checks work on each XML file
individually, so don't have the schema information available when looking at
the data in the xcu file.  And those checks are written in XSLT, which makes
writing new checks rather, erm, unpleasant.)

Change-Id: I9c61e8ebf75d3d54f7e5b6688eac47a6ed8dc5c3
Reviewed-on: https://gerrit.libreoffice.org/49252
Tested-by: Jenkins <ci@libreoffice.org>
Reviewed-by: Stephan Bergmann <sbergman@redhat.com>
---
 officecfg/registry/schema/org/openoffice/Setup.xcs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/officecfg/registry/schema/org/openoffice/Setup.xcs b/officecfg/registry/schema/org/openoffice/Setup.xcs
index f55e3698f58f..31db1eca9af4 100644
--- a/officecfg/registry/schema/org/openoffice/Setup.xcs
+++ b/officecfg/registry/schema/org/openoffice/Setup.xcs
@@ -149,7 +149,7 @@
           <desc>Specifies the short name of the factory.</desc>
         </info>
       </prop>
-      <prop oor:name="ooSetupFactoryUIName" oor:type="xs:string">
+      <prop oor:name="ooSetupFactoryUIName" oor:type="xs:string" oor:localized="true">
         <!-- UIHints: setup -->
         <info>
           <desc>Specifies the UI name of the factory.</desc>
-- 
2.14.3

